source/module_base/test/tool_check_test.cpp

```diff
@@ -0,0 +1,98 @@
+#include "../tool_check.h"
+#include "gtest/gtest.h"
+#include <fstream>
+#include <cstdio>
+
+/************************************************
+*  unit test of class ToolCheckTest
+***********************************************/
+
+/**
+ * - Tested Function
+ *   - ModuleBase::CHECK_NAME
+ *	- check the next input from ifs is std::string
+ *
+ *   - ModuleBase::CHECK_INT
+ *	- check the next input from ifs is int
+ *
+ *   - ModuleBase::CHECK_DOUBLE
+ *	- check the next input from ifs is double
+ *
+ *   - ModuleBase::CHECK_STRING
+ *	- check the next input from ifs is string
+ *
+ */
+
+
+class ToolCheckTest : public testing::Test
+{
+    protected:
+        std::ofstream ofs;
+	std::ifstream ifs;
+	// define std::string, int, double variables
+	std::string name="abaqus";
+	int ecut = 100;
+	double occupation = 0.23;
+	std::string caltype="nscf";
+	// quit is the swith to control performance of function
+	bool quit=false;
+};
+
+TEST_F(ToolCheckTest,Name){
+	ofs.open("tmp");
+	// double input to check continus check function
+	ofs << name << std::endl;
+	ofs << name << std::endl;
+	ofs.close();
+	ifs.open("tmp");
+	// non-quit check
+        ModuleBase::CHECK_NAME(ifs,"abacus",quit);
```

Comment:
Checks on return value of this function should be performed. Same below.

Reply:
done